barretenberg/cpp/src/barretenberg/vm2/simulation/lib/u256_decomposition.cpp

```diff
@@ -0,0 +1,14 @@
+#include "barretenberg/vm2/simulation/lib/u256_decomposition.hpp"
+
+namespace bb::avm2::simulation {
+
+const uint256_t TWO_POW_128 = uint256_t{ 1 } << 128;
```

Comment:
nit: prefer `uint256_t(1)`. Brace initialization is a can of worms. I like to follow this: https://abseil.io/tips/88